code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -1,56 +1,165 @@
+		if(owner_xeno.do_actions)
+			return FALSE
+		if(!HAS_TRAIT(target_human, TRAIT_UNDEFIBBABLE))
+			if(!silent)
+				to_chat(owner_xeno, span_notice("We can only use fully dead hosts to heal."))
+			return FALSE
+		if(target_human.blood_volume < GORGER_REJUVENATE_BLOOD_DRAIN)
+			if(!silent)
+				to_chat(owner_xeno, span_notice("Our meal has no blood... How sad!"))
+			return FALSE
+		return TRUE
+
+	if(!.)
+		return
+
+	if(owner_xeno.plasma_stored >= owner_xeno.xeno_caste.plasma_max)
 		if(!silent)
-			to_chat(X, span_xenowarning("It is outside of our reach! We need to be closer!"))
+			to_chat(owner_xeno, span_xenowarning("No need, we feel sated for now..."))
 		return FALSE
-/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/A)
-	var/mob/living/carbon/xenomorph/X = owner
-	X.face_atom(A)
-	X.do_attack_animation(A, ATTACK_EFFECT_GRAB)
-	X.emote("roar")
-	A.drop_all_held_items()
-	X.visible_message(A, span_danger("The [X] grabs [A]!"))
-	A.SetImmobilized(2 SECONDS)
-	if(!do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE, ignore_turf_checks = FALSE))
-		to_chat(X, span_xenowarning("Our meal is interrupted!"))
-		succeed_activate()
-		add_cooldown()
+
+/datum/action/xeno_action/activable/drain/use_ability(mob/living/carbon/target)
+	var/mob/living/carbon/xenomorph/owner_xeno = owner
+	if(target.stat == DEAD)
+		var/mob/living/carbon/human/target_human = target
+		while(owner_xeno.health < owner_xeno.maxHealth && do_after(owner_xeno, 2 SECONDS, TRUE, target_human, BUSY_ICON_HOSTILE))
+			if(target_human.blood_volume < GORGER_REJUVENATE_BLOOD_DRAIN)
+				to_chat(owner, span_notice("Our meal has no blood... How sad!"))
+				return
+			owner_xeno.heal_wounds(2.2, TRUE)
+			owner_xeno.adjust_sunder(-0.5)
+			target.blood_volume -= GORGER_REJUVENATE_BLOOD_DRAIN
+		to_chat(owner_xeno, span_notice("We feel fully restored."))
 		return
-	X.do_attack_animation(A, ATTACK_EFFECT_REDSTAB)
-	A.emote("scream")
-	A.drop_all_held_items()
-	X.visible_message(A, span_danger("The [X] stabs its tail into [A]!"))
-	A.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
-	playsound(A, "alien_claw_flesh", 25, TRUE)
-	X.gain_plasma(X.xeno_caste.drain_plasma_gain)
+	owner_xeno.face_atom(target)
+	owner_xeno.emote("roar")
+	target.drop_all_held_items()
+	owner_xeno.do_attack_animation(target, ATTACK_EFFECT_REDSTAB)
+	owner_xeno.visible_message(target, span_danger("The [owner_xeno] stabs its tail into [target]!"))
+	playsound(target, "alien_claw_flesh", 25, TRUE)
+	target.emote("scream")
+	target.apply_damage(damage = 20, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD, blocked = 0, sharp = TRUE, edge = FALSE, updating_health = TRUE)
+	target.ParalyzeNoChain(0.25 SECONDS)
+	target.blur_eyes(1)
+	owner_xeno.gain_plasma(owner_xeno.xeno_caste.drain_plasma_gain)
 	succeed_activate()
 	add_cooldown()
 
+/datum/action/xeno_action/activable/drain/ai_should_use(atom/target)
```

Comment:
you just need to return can_use_ability in that proc, nothing else.

Reply:
changed